File: StudiumAnalytics-stable/spark/apps/studium-analytics-streaming-1.0.0/src/main/java/studium/analytics/streaming/Models/TeachingState.java
package studium.analytics.streaming.Models;


import java.io.Serializable;

import com.google.gson.*;
import java.sql.Timestamp;


public class TeachingState implements Serializable {
    private static final long serialVersionUID = 1L;
    public Teaching teaching;
    public Boolean showCaseExists;
    public Boolean notifiesExists ;
    public Boolean descriptionExists;
    public Boolean documentsExists ;
    public Boolean bookingExists;   
    
    public String toString()
    {
        Gson serializer = new Gson();
        String teachStateJson = serializer.toJson(this);
        String s1 = "{\"timestamp\" : "+new Timestamp(System.currentTimeMillis()).getTime()+",";
        s1 += teachStateJson.substring(1, teachStateJson.length());
        return s1;
    }
}

File: StudiumAnalytics-stable/dotnetBackend/StudiumAnalytics/StudiumAnalyticsBackend/Models/TeachingState.cs
using System;
using System.Text.Json;

namespace StudiumAnalyticsBackend.Models
{
    public class TeachingState
    {
        public Teaching teaching { get; set; }

        public bool showCaseExists { get; set; }

        public bool notifiesExists { get; set; }

        public bool descriptionExists { get; set; }

        public bool documentsExists { get; set; }

        public bool bookingExists { get; set; }

        

        public override string ToString()
        {
            return JsonSerializer.Serialize<TeachingState>(this);
        }
    
    }
}
